operator/api/pulumi/v1/program_types.go

```diff
@@ -160,16 +160,35 @@ type Getter struct {
 	State map[string]Expression `json:"state,omitempty"`
 }
 
+// StackStatus defines the observed state of Program.
+type ProgramStatus struct {
+	// ObservedGeneration is the last observed generation of the Program
+	// object.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// LastResyncTime contains a timestamp for the last time a resync of the stack took place.
+	// +optional
+	LastResyncTime metav1.Time `json:"lastResyncTime,omitempty"`
```

Comment:
I still think this field should be removed because a) it is redundant given the observedGeneration, and b) it uses the term "sync" which has a particular (different) meaning in PKO.

Reply:
- `field should be removed`: I'm opposed to this more so because we can have multiple reconciliation on the same generation. Take the example where a Program object has been reconciled and the status is written to it. If the controller pod restarts, there might be a possibility that the hostname has changed (though very unlikely). When the controller pod restarts, it will reconcile all objects it should watch for, and thus, the URL stored in status will be updated. It would be helpful for debugging purposes to know when the URL was generated.
- `the term "sync" which has a particular (different) meaning in PKO.`: yes, a different name for this field, should we ultimately decide to keep it, should be used!

Reply:
I've removed this field for now, to merge this PR in eagerly. We can rethink how we want to handle this and add this in the future if we choose to.